### PR TITLE
Major refactor of markup, styles and components for home page

### PR DESCRIPTION
**Pull request by huijing**
Part of https://github.com/WICG/webmonetization/issues/320

This PR hits quite a number of files (although about half are due to image optimisation). The most significant changes are to the custom.css file and the index.js file in their entireties.

A summary of the intention for this endeavour is to properly utilise the libraries that we have chosen to include in the site, e.g. Material UI, as well as to introduce more consistency in markup patterns across our 5 custom pages. This PR mainly focuses on the index.js file, but tweaks some of the other files due to the removal/refactor of CSS.

This PR introduces:
- Responsive typographic sizing (generated from https://utopia.fyi/), therefore the font sizes will look different (implemented via overriding of Infima variables)
- Responsive spacing, also from https://utopia.fyi/ (implemented via overriding of Infima variables)
- A complete rewrite of the home page markup to better utilise the available components from Material UI where appropriate, otherwise, hand-rolled CSS where better suited
- Optimization of all SVG images used on the home page, and replacement of PNGs with SVGs instead
- Fixes some of the existing layout issues (i.e. layout is overflowing at some narrower widths)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,130 +28,6 @@ $ yarn
 $ yarn start
 ```
 
-### Directory Structure
```
Comment (huijing) [inline README.md:31]: Given the amount of changes made to the website, this section becomes obsolete real quick, so we could probably just remove it.

File: docusaurus.config.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ const tools = [
   },
   {
     name: 'Vuepress Plugin Web-Monetization',
-    image: 'logo-tool-vuepress.png',
+    image: 'logo-tool-vuepress.svg',
```
Comment (huijing) [inline docusaurus.config.js:62]: SVGs scale better and easier to manage.

File: src/page-support/exclusive-content/components/copy-content.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import {
   DialogContentText,
   Tooltip,
 } from '@mui/material'
-import { FileCopyOutlinedIcon, CheckCircleIcon } from '@mui/icons-material'
+import { FileCopyOutlined, CheckCircle } from '@mui/icons-material'
```
Comment (huijing) [inline src/page-support/exclusive-content/components/copy-content.js:10]: Due to the MaterialUI version bump, the name of the icons have been updated.

File: src/page-support/exclusive-content/components/generated-tabs.js [hunk 4]
```diff
@@ -14,7 +14,11 @@ export function GeneratedTabs() {
   return (
     <>
       <div>
-        <AppBar position='static' color='none' style={{ boxShadow: 'none' }}>
+        <AppBar
+          position='static'
+          color='transparent'
```
Comment (huijing) [inline src/page-support/exclusive-content/components/generated-tabs.js:19]: Syntax change from Material UI upgrade

File: src/page-support/prob-revshare/components/active-view.js [hunk 5]
```diff
@@ -10,32 +10,34 @@ import { ImportText } from './import-text'
...
-      <OnlyClient>
-        <RevshareChart />
-        <ShareList />
-        <SectionHeader>
-          Meta Tag
-        </SectionHeader>
-        <SharesMetaTag />
-      </OnlyClient>
-    </>
+    return (
```
Comment (huijing) [inline src/page-support/prob-revshare/components/active-view.js:17]: Auto-formatting has occured. Apologies.

File: src/pages/exclusive-content.js [hunk 6]
```diff
@@ -16,25 +16,21 @@ export default function ExclusiveContent(props) {
     <Layout
       permalink='/exclusive-content'
       title={siteConfig.title}
-      description={siteConfig.tagLine}
+      description={siteConfig.tagline}
```
Comment (huijing) [inline src/pages/exclusive-content.js:19]: Turns out we spelled this wrong the whole time. 🤨

File: src/pages/exclusive-content.js [hunk 7]
```diff
@@ -16,25 +16,21 @@ export default function ExclusiveContent(props) {
...
-                This tool allows you to create exclusive content which is only
-                unlocked if payment can be verified.{' '}
-                <Link to='/docs/exclusive-content#exclusive-content-with-payment-verification'>
-                  Learn more!
-                </Link>
-              </p>
-              <ActiveView />
-            </Container>
-          </div>
+          <Container maxWidth='md' component='section'>
```
Comment (huijing) [inline src/pages/exclusive-content.js:23]: Removal of unnecessary wrappers

File: src/pages/index.js [hunk 8]
```diff
@@ -1,325 +1,253 @@
 import React from 'react'
-import { Container } from '@mui/material'
+import { Container, Grid } from '@mui/material'
```
Comment (huijing) [inline src/pages/index.js:2]: Used for the Resource section. Was not used for the cards simply because the Grid API for Material UI does not allow us to use the `a` element for Grid items and this impacts the shadow styling and hover effects implementation. As such, the cards are hand-rolled instead.

File: src/pages/index.js [hunk 9]
```diff
@@ -1,325 +1,253 @@
...
 import Layout from '@theme/Layout'
+import Link from '@docusaurus/Link'
```
Comment (huijing) [inline src/pages/index.js:4]: For the internal links in the resources section

File: src/pages/index.js [hunk 10]
```diff
@@ -1,325 +1,253 @@
...
 
-class HomeSplash extends React.Component {
-  render() {
-    const { siteConfig } = this.props
-    const { baseUrl } = siteConfig
+export default function Index(props) {
```
Comment (huijing) [inline src/pages/index.js:6]: Rewrite to use the functional pattern over the old class pattern

**Review by sublimator — APPROVED** [review]

File: src/pages/index.js [hunk 11]
```diff
@@ -1,325 +1,253 @@
...
-      </TextBlock>
-    )
-
-    const WhyNow = () => (
-      <TextBlock>
-        <h2>Why Now?</h2>
-        <p>
-          Until recently, there hasn't been an open, neutral protocol for
-          transferring money.{' '}
+  const CardList = (props) => {
```
Comment (huijing) [inline src/pages/index.js:9]: Simplified the component structure to just having `CardList` to deal with the card data pulled from docusaurus.config.js and `PrCta` which is repeated throughout the sections asking folks to open a PR for proposed suggestions.

File: src/pages/index.js [hunk 12]
```diff
@@ -1,325 +1,253 @@
...
-              <br />
-              <a href={docUrl('docs/getting-started', this.props.language)}>
-                Read docs ›
-              </a>
-            </p>
-          </div>
-          <div className='resource'>
-            <h3>Explainer</h3>
+  return (
+    <Layout
```
Comment (huijing) [inline src/pages/index.js:66]: Markup has been greatly simplified to reduce redundant wrappers.
Due to the design of the site's hero section with a full bleed background, `Container` is used as an inner wrapper instead to keep the content within a specific max-width. The max-width values out-of-the-box from Material UI are fairly reasonable and suit the type of content we have, so that's what we're using now.

File: src/pages/meta-tag.js [hunk 13]
```diff
@@ -32,74 +33,66 @@ export default function MetaTag(props) {
...
-                src='/img/copy_icon.svg'
-                id='copyIcon'
-                className='btnClipboard'
-                alt='copy-icon'
-              />
-            </code>
-          </div>
-        </Container>
-      </div>
+          <CodeBlock className='language-html'>
```
Comment (huijing) [inline src/pages/meta-tag.js:91]: Docusaurus does allow for use of the built-in default code syntax highlighter in custom pages as well, and we should make use of this instead of rolling our own.
See https://github.com/facebook/docusaurus/issues/6185

File: src/pages/meta-tag.js [hunk 14]
```diff
@@ -1,7 +1,8 @@
 import React, { useState } from 'react'
-import { Container, TextField } from '@mui/material'
+import { Container, TextField, Button } from '@mui/material'
```
Comment (huijing) [inline src/pages/meta-tag.js:2]: Use of the Button component will be assessed in the PR specific to this Link Tag Generator page

File: src/pages/prob-revshare.js [hunk 15]
```diff
@@ -16,25 +16,21 @@ export default function ProbRevshare(props) {
...
-                This tool allows you to create a Web Monetization meta tag which
-                splits money between multiple Payment Pointers (
-                <Link to='/docs/probabilistic-rev-sharing'>Learn more</Link>).
-                If you make changes to an existing meta tag with this tool, make
-                sure to replace the old meta tag with the updated version.
-              </p>
-              <ActiveView />
-            </Container>
-          </div>
+          <Container maxWidth='md' component='section'>
```
Comment (huijing) [inline src/pages/prob-revshare.js:23]: Again, part of the markup refactor. Further refactoring of this page specifically will be in a separate PR.

File: static/css/custom.css [hunk 16]
```diff
@@ -1,54 +1,82 @@
 :root {
+  --step--1: clamp(0.8rem, calc(0.7rem + 0.44vw), 0.95rem);
```
Comment (huijing) [inline static/css/custom.css:2]: The variables without the `--ifm` namespace are related to the responsive typography and spacing from Utopia.fyi, while those with the namespace are to override the default options.

File: static/css/custom.css [hunk 17]
```diff
@@ -1,54 +1,82 @@
...
+  --ifm-global-spacing: var(--space-s);
+  --ifm-footer-link-color: var(--ifm-color-emphasis-300);
 }
 
-/* custom css */
-h2 {
-  font-size: 40px;
-  margin-top: 60px;
-  margin-bottom: 16px;
+/* Typography styles */
```
Comment (huijing) [inline static/css/custom.css:37]: The refactoring essentially tries to reduce magic numbers as much as possible, and standardise values using the provided variables. Also introducing comments as delimiters for styles specific to certain pages or the overall site such that it is easier to maintain.

File: static/css/custom.css [hunk 18]
```diff
@@ -58,281 +86,181 @@ h2 {
...
-  padding: 18px 50px 18px 24px;
-  display: block;
-  line-height: 2em;
-  position: relative;
-  background: #f7f8f9;
+.icon--merge {
+  width: 1em;
 }
 
+/* Link tag generator page styles */
```
Comment (huijing) [inline static/css/custom.css:261]: These will be dealt with in each respective page's refactor in future fast-follow PRs.

File: static/css/custom.css [hunk 19]
```diff
@@ -58,281 +86,181 @@ h2 {
...
-.card-wide-content h4 {
-  margin: 0 0 8px 0;
-  font-size: 17px;
-  color: #000;
+[class*='home-card--'] {
+  padding: var(--space-s);
 }
 
-.card-wide-arrow {
+[class*='home-card--']::after {
```
Comment (huijing) [inline static/css/custom.css:188]: This is a performance-related choice, see: https://tobiasahlin.com/demo/animate-box-shadow/

**Comment by sublimator** [comment]
LGTM
